src/Parsing/Parser.hs

```diff
@@ -62,26 +77,26 @@ import           TreeSitter.Unmarshal
 
 
 -- | A parser, suitable for program analysis, for some specific language, producing 'Term's whose syntax satisfies a list of typeclass constraints.
-data SomeAnalysisParser typeclasses ann where
-  SomeAnalysisParser :: ( ApplyAll typeclasses (Sum fs)
+data SomeAnalysisParser constraint ann where
```

Comment:
Explicit `:: Constraint` kind signature on `constraint` here, just for safety’s sake?